test/github.test.js

```diff
@@ -63,17 +75,141 @@ describe("Github plugin", function() {
+         cb++;
+
+         var sanitizedBuild = _.pick(build, 'repo', 'commit', 'branch', 'status')
+         assert.deepEqual(sanitizedBuild, {
+            repo:    'github.com/user/repo',
+            commit:   pullRequestCommit,
+            branch:  pullRequestTestBranchName,
+            status:  'pending'
+         });
+      finished();
+
+      // Ensure this callback doesn't get called again by
+      // delaying the done()
+      setTimeout(function() {
+         cimpler.shutdown();
+         assert.equal(cb, 1, "Assert the callback was called once.");
+         done();
+         }, 100);
+      });
+   });
+   it("should listen for and add builds for pull_request opened events",
+   function(done) {
+      // Set event type to 'pull_request' and action to 'opened'
+      options.headers["X-GitHub-Event"] = "pull_request";
+      postBuild.action = 'opened';
+
+      var cimpler = new Cimpler({
+         httpPort: httpPort
+      });
+
+      cimpler.registerPlugin(Github, { });
+
+      var check = expect(1, function(count) {
+         assert.fail();
+      });
+
+      var req = http.request(options);
+
+      // Pretend to be github and POST to our listener
+      req.write("payload=" + encodeURIComponent(JSON.stringify(postBuild)));
+      req.end();
+
+      var cb = 0;
+      cimpler.consumeBuild(function(build, started, finished) {
+         assert.equal(cb, 0, "Only one build should get through");
+         cb++;
+
+         var sanitizedBuild = _.pick(build, 'repo', 'commit', 'branch', 'status')
+         assert.deepEqual(sanitizedBuild, {
+            repo:    'github.com/user/repo',
+            commit:   pullRequestCommit,
+            branch:  pullRequestTestBranchName,
             status:  'pending'
          });
          finished();
-         // Ensure this callback doesn't get called again by
-         // delaying the done()
+
+      });
+      // Ensure this callback doesn't get called again by
+      // delaying the done()
       setTimeout(function() {
          cimpler.shutdown();
+         assert.equal(cb, 1, "Assert the callback was called once.");
          done();
       }, 100);
+   });
+   it("should listen but abort builds for pull_requests with actions other than synchronize and opened",
+   function(done) {
+      // Set event type to 'pull_request' and action to 'closed'
+      options.headers["X-GitHub-Event"] = "pull_request";
+      postBuild.action = 'closed';
+
+      var cimpler = new Cimpler({
+         httpPort: httpPort
+      });
+
+      cimpler.registerPlugin(Github, { });
+
+      var check = expect(1, function(count) {
+         assert.fail();
+      });
+
+      var req = http.request(options);
+
+      // Pretend to be github and POST to our listener
+      req.write("payload=" + encodeURIComponent(JSON.stringify(postBuild)));
+      req.end();
+
+      var cb = 0;
+      cimpler.consumeBuild(function(build, started, finished) {
```

Comment:
```suggestion
      cimpler.consumeBuild(function() {
```

I don't really know how this works, but it seems like these params aren't used.

Reply:
We don't need it because we bail before we would use those variables.